src/participant/participant.ts

```diff
@@ -811,50 +784,205 @@ export default class ParticipantController {
     };
   }
 
-  async _askForNamespace({
-    command,
+  async _getDatabases({
+    stream,
+  }: {
+    stream: vscode.ChatResponseStream;
+  }): Promise<
+    | {
+        _id: string;
+        name: string;
+      }[]
+    | undefined
+  > {
+    stream.push(
+      new vscode.ChatResponseProgressPart('Fetching database names...')
+    );
+    const dataService = this._connectionController.getActiveDataService();
+    if (!dataService) {
+      return;
+    }
+
+    try {
+      const databases = await dataService.listDatabases({
+        nameOnly: true,
+      });
+      return databases;
+    } catch (error) {
+      log.error('Unable to fetch databases:', error);
+    }
+  }
+
+  async _getCollections({
+    stream,
+    databaseName,
+  }: {
+    stream: vscode.ChatResponseStream;
+    databaseName: string;
+  }): Promise<ReturnType<DataService['listCollections']> | undefined> {
+    stream.push(
+      new vscode.ChatResponseProgressPart('Fetching collection names...')
+    );
+
+    const dataService = this._connectionController.getActiveDataService();
+
+    if (!dataService) {
+      return;
+    }
+    try {
+      return await dataService.listCollections(databaseName);
+    } catch (error) {
+      log.error('Unable to fetch collections:', error);
+    }
+  }
+
+  /** Gets the collection name if there is only one collection.
+   *  Otherwise returns undefined and asks the user to select the collection. */
+  async _getOrAskForCollectionName({
     context,
     databaseName,
-    collectionName,
     stream,
+    command,
   }: {
     command: ParticipantCommand;
     context: vscode.ChatContext;
-    databaseName: string | undefined;
-    collectionName: string | undefined;
+    databaseName: string;
     stream: vscode.ChatResponseStream;
-  }): Promise<ChatResult> {
+  }): Promise<string | undefined> {
+    const collections = await this._getCollections({ stream, databaseName });
+
+    if (collections === undefined) {
+      return;
+    }
+    
+    if (collections.length === 1) {
+      return collections[0].name;
+    }
+
+    stream.markdown(
+      vscode.l10n.t(
+        `Which collection would you like to use within ${databaseName}?\n\n`
+      )
+    );
```

Comment:
If there are no collections in the database, would the user have to type in the collection name in the chat themselves?

Reply:
yeah. when they type nothing afterwards we actually send a different text `Please select a collection by either clicking on an item in the list or typing the name manually in the chat` which I feel like is a better text for this in general, maybe we could just use that?